read-fonts/src/tables/glyf.rs

```diff
@@ -477,6 +478,175 @@ impl<'a> SomeTable<'a> for Component {
     }
 }
 
+/// Errors that can occur when converting an outline to a path.
+#[derive(Clone, Debug)]
+pub enum ToPathError {
+    /// Contour end point at this index was less than its preceding end point.
+    ContourOrder(usize),
+    /// Expected a quadratic off-curve point at this index.
+    ExpectedQuad(usize),
+    /// Expected a quadratic off-curve or on-curve point at this index.
+    ExpectedQuadOrOnCurve(usize),
+    /// Expected a cubic off-curve point at this index.
+    ExpectedCubic(usize),
+}
+
+impl fmt::Display for ToPathError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::ContourOrder(ix) => write!(
+                f,
+                "Contour end point at index {ix} was less than preceeding end point"
+            ),
+            Self::ExpectedQuad(ix) => write!(f, "Expected quadatic off-curve point at index {ix}"),
+            Self::ExpectedQuadOrOnCurve(ix) => write!(
+                f,
+                "Expected quadatic off-curve or on-curve point at index {ix}"
+            ),
+            Self::ExpectedCubic(ix) => write!(f, "Expected cubic off-curve point at index {ix}"),
+        }
+    }
+}
+
+/// Converts a `glyf` outline described by points, flags and contour end points to a sequence of
+/// path elements and and invokes the appropriate callback on the given sink for each.
+///
+/// The input points are expected in `F26Dot6` format as that is the standard result of scaling
+/// a TrueType glyph. Output points are generated in `f32`.
+pub fn to_path(
+    points: &[Point<F26Dot6>],
+    flags: &[u8],
```

Comment:
are these `SimpleGlyphFlags`? some other flag type? Would this code benefit from having that type be defined?

Reply:
This is a bit complicated. For the purposes of `to_path`, they would only have the on-curve (or a prospective cubic off-curve) bit set. But this array is overloaded and is used by both variation delta computation and hinting to store temporary flags, specifically for IUP processing. It could potentially be typed, but I'm not sure exactly what that might look like or if it would be helpful.

Reply:
the main way I would imagine it being helpful would be as types-as-documentation; currently I look at this and I see a byte array and it could be anything. If it were a type you would also be able to add some simple convenience methods for it for testing various bits, which might also let you make this whole method slightly more readable? I'm fine either way, but if you want to keep it like this it might be nice to document somewhere what these flags are, exactly.

Reply:
Sold. I’ll add a `PointFlags` type with methods for the various bits.